cmd/new.go

```diff
@@ -367,6 +371,29 @@ func errorIfNotEmptyDirectory(path string) error {
 	return nil
 }
 
+func validateProjectName(projectName string, opts display.Options) error {
+	err := workspace.ValidateProjectName(projectName)
+	if err != nil {
+		return err
+	}
+
+	b, err := currentBackend(opts)
+	if err != nil {
+		return err
+	}
+
+	exists, err := b.DoesProjectExist(commandContext(), projectName)
```

Comment:
If the `--generate-only` (or `-g`) flag is specified, we won't be creating a stack. I wonder if in that case we should just skip getting the backend and calling `b.DoesProjectExist`?

Reply:
But the project name still needs to be unique in this case, doesn't it?

Reply:
> But the project name still needs to be unique in this case, doesn't it?

It does, but when `-g` is specified, no actual stack is going to be created as part of the `pulumi new` command invocation. I suppose it is likely they'll run `pulumi stack init` as a next step at some point (or indirectly via running `pulumi up` and it prompting to create a stack). And having `new` check to ensure the project is unique before any of these subsequent commands could be helpful. However, by the time they run the command that creates the stack, the projects on the server _could_ have changed (although unlikely), or they'll want to create a stack in a different org, in which case the check in the user's org was pointless and would artificially block them when they just want to emit the project on disk.

So I guess I lean slightly towards not doing the check when `-g` is specified. It's not creating any stacks, so don't bother with with the uniqueness check. But I could be convinced otherwise.